crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -365,6 +408,56 @@ impl<'db> TypeInferenceBuilder<'db> {
         self.types.definitions.insert(definition, function_ty);
     }
 
+    fn infer_parameters(&mut self, parameters: &ast::Parameters) {
+        let ast::Parameters {
+            range: _,
+            posonlyargs,
+            args,
+            vararg,
+            kwonlyargs,
+            kwarg,
+        } = parameters;
+
+        for param_with_default in posonlyargs {
+            self.infer_parameter_with_default(param_with_default);
+        }
+        for param_with_default in args {
+            self.infer_parameter_with_default(param_with_default);
+        }
+        if let Some(vararg) = vararg {
+            self.infer_parameter(vararg);
+        }
+        for param_with_default in kwonlyargs {
+            self.infer_parameter_with_default(param_with_default);
+        }
+        if let Some(kwarg) = kwarg {
+            self.infer_parameter(kwarg);
+        }
```

Comment:
Nit: I think `parameters` implements `Iterator` which returns all parameters in the right order.